### PR TITLE
Luliya

### PR DESCRIPTION
**Pull request by LyleSY**
#### Summary
Content "Exodii NPC Luliya"

#### Purpose of change

Flesh out Exodii, make base more interesting

#### Describe the solution

New NPC Luliya with Exodii faction, spawns on the floor above Rubik at the exodii base with a broken exodii quad and a golden krar and golden body. She doesn't speak English but eventually the player can figure out that she likes music and if the player brings an instrument she can sing and play with them and give a small morale boost.  Some dialogue is gated by time so there will be different interactions at different visits. I believe this is the first time we've used "[You see a thing and then you do a thing]" format in NPC dialogue, but I think it's necessary as a workaround here and also I think it works well and may make sense elsewhere.

#### Describe alternatives you've considered

Stuff I'm definitely not doing:

- Give her a broom or medical gloves. 
- Possibility to make her a follower.  
- Give the player the option to kick the body before it's been activated. 

Stuff that may be good to do in the future:

- Dialogue for Rubik about her. (highest priority and easiest) 
- I'd like to add some dialogue at some point for the player to teach her English to learn more about her deal. I also like the idea to get some kind of translator hack from Rubik. Alternately she could just teach herself passively if the survivor lives long enough.
- Explain her backstory which I've thought way too much about.
- Dialogue for her about Rubik.
- Patrik's idea of a quest chain to fix the quad is extremely interesting. I'm thinking that could be her brother who escaped with her, Samson. 
- Give her a quest or service to find MP3 players or music files in exchange for something.
- Other quests for the player to get various unique materials, possibly in exchange for faction credit. This could be a way to drive the player to unique content that is otherwise not connected to any missions.
- Quest chain to bump her up to a better body. 
- Quest chain to cure her wibbly-wobblies and allow her to reconnect to the other exodii. 
- Quest chain for the player to get her golden body or a golden krar.

Stuff there isn't code support for yet:

- Let her order quads to clean CBMs for the player for free with a small delay.
- Dating and romance options. Marriage, 2.5 cyborg kids, house in the virtual suburbs, cyborg dog.

#### Testing

Game loads no errors, map assets and Luliya spawn. Dialogue works as expected.
Screenshot 2023-03-10 at 10 55 17 PM

#### Additional context

Thanks to Erk for talking this through with me.

### CODE REVIEW

File: data/json/npcs/exodii/exodii_Luliya.json [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+  {
+    "type": "talk_topic",
+    "id": "TALK_LULIYA_BUTTON",
+    "dynamic_line": "<You hear a loud static discharge, a rough mechanical chugging sound, and then high-pitched shrieking with strange distortions and modulations in the sound.  The golden figure shudders, arms flailing> ቆዳዬ! ቆዳዬ በእሳት ላይ ነው! እየሞትኩ ነው!",
+    "speaker_effect": { "effect": { "u_add_var": "u_met_luliya", "type": "general", "context": "meeting", "value": "yes" } },
+    "responses": [
+      { "text": "Hello.  Are you all right?  I just hit the button on your back.", "topic": "TALK_LULIYA" },
+      { "text": "I'm not understanding that.  Can you try again?", "topic": "TALK_LULIYA" },
+      {
+        "text": "Speak American robot.  We don't take kindly to that kind of thing here.  <You frown menacingly>",
```
Comment (MNG-cataclysm) [inline data/json/npcs/exodii/exodii_Luliya.json:124]:
```suggestion
        "text": "Speak English ya <swear> robot.  We don't take kindly to that kind of thing here.  <You frown menacingly>",
```

File: data/json/npcs/exodii/exodii_Luliya.json [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        "topic": "TALK_LULIYA_MUSIC2",
+        "condition": { "u_has_item": "guitar" }
+      },
+      {
+        "text": "[HAS TRUMPET] I happen to have a nice trumpet, want to hear something?",
+        "topic": "TALK_LULIYA_MUSIC2",
+        "condition": { "u_has_item": "trumpet" }
+      },
+      {
+        "text": "[HAS SAXOPHONE] I happen to have a nice trumpet, want to hear something?",
```
Comment (MNG-cataclysm) [inline data/json/npcs/exodii/exodii_Luliya.json:198]:
```suggestion
        "text": "[HAS SAXOPHONE] I happen to have a nice saxophone, want to hear something?",
```

**Review by MNG-cataclysm — CHANGES_REQUESTED** [review]
Choosing to kick Luliya repeatedly should probably cause her to become upset with the player, rather than doing nothing. 

Also, what if she could use a mechanical translator similar to that of Rubik, which he uses to explain things if the player doesn’t understand something? It seems that both could access such technology, or the Great Gray for the purposes of translating.

<P.S, If she does become a possible romance, choosing to play the saxophone in the music dialogue should make any related checks automatically pass in your favor.>

**Comment by PatrikLundell** [comment]
I'm not sure romance with an Exodii cyborg would make sense, as the body parts that would have been involved are likely sacrificed and replaced with robotic parts that lack the now unneeded capabilities.

I'm also not sure the Exodii would be too keen to lose some of their limited staff on one of their temporary stops, unless sufficiently payed for in an increased trade and a decent chance of recruitment of suitable locals into their faction. However I don't think there's any system for an NPC to influence the player to do what factions want, beyond punishment/rewards.

That setting might allow for quests where the player provides Luliya with items used to repair the quad. That could be done through a "dialog" where hand writing and drawing are used to convey the message of what's needed and what the reward would be, possibly gated behind skill checks to understand what the parts needed are (and an option to guess what they might be if the skill isn't sufficient, although that might lead to meta gaming). It might be possible to make a quest chain where language gradually is introduced, to simulate the learning of a local language.

**Comment by MNG-cataclysm** [comment]
> I'm not sure romance with an Exodii cyborg would make sense, as the body parts that would have been involved are likely sacrificed and replaced with robotic parts that lack the now unneeded capabilities.

I suppose romance with the Exodii could occur through the Great Gray or other networks, relying a dream-like trance between two or more cyborgs. Otherwise, there could be certain bionic modules that replace the parts needed with mechanical equivalents, or augment them with some sort of *enhancement*. It is one of the most marketable things around, it's possible one or more worlds the Exodii visited had something like that available.

Considering that Lulia is described as vain, she probably would want to install one or more of these.

**Comment by Maddremor** [comment]
Remember that outsiders are ephemeral and disposable. I doubt anyone that formed strong emotional attachments that easily could cope with the Exodii lifestyle.

**Comment by harakka** [comment]
> Considering that Lulia is described as vain, she probably would have installed one or more of these.

That is hell of a conclusion to jump to.

> Considering that Lulia is described as vain, she probably would have installed one or more of these.

That is, uh, quite a conclusion to jump into.

**Comment by MNG-cataclysm** [comment]
> That is hell of a conclusion to jump to.

Would the mechanical dream-trance work better for this?

**Comment by Maddremor** [comment]
I don't think sex and eroticism is especially interesting or relevant here.

**Comment by I-am-Erk** [comment]
I'd like to leave any concept of romancing a cyborg from another dimension in the vague clouds of your imagination for the moment pls.

Exodii are not really good candidates to become followers.

**Comment by kevingranade** [comment]
The exodii do not need a cleaning lady. They have the resources for a better roomba if they want that, they definitely don't need to dedicate a member and their highly capable robot body to that task.
The kicking options are right out,  if they're not meaningful interactions the player can feel free to imagine doing it.

**Comment by MNG-cataclysm** [comment]
What if Luilya oversaw material recovery and management, instead of cleaning?

**Comment by fairyarmadillo** [comment]
Putting aside romance, which I don't think the game is set up to handle, a long-running questline/plot where the player forms a bond with a member of a faction as remote as the Exodii is a pretty cool idea. It probably happens frequently enough that a member of the faction will get inappropriately (from the faction's point of view) attached to an outsider which leads to some angst when it comes time to jump to a new world and leave said outsider behind.

A story like that could easily be expressed through dialog with Rubik and Luliya, especially if the player ever gets a language implant or proficiency to communicate with the latter. The current gameplay loop encourages re-visiting this faction frequently, so giving them more things to say and do over time might make this more interesting.

File: data/json/npcs/exodii/exodii_Luliya.json [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+  {
+    "type": "talk_topic",
+    "id": "TALK_LULIYA",
+    "dynamic_line": {
+      "u_has_var": "u_met_luliya",
+      "type": "general",
+      "context": "meeting",
+      "value": "yes",
+      "yes": [ "Inikwani dehina metahi tegwazhi." ],
+      "no": "[You see a tall golden figure in the closet with female features.  Strapped to the back of the head with copper wire is a glass cylinder containing long frizzy black hair.  The body stands still and emits an unpleasant beeping sound.  On the back there is a large red button with strange markings.  There is a golden harp of some kind.]"
```
Comment (MNG-cataclysm) [inline data/json/npcs/exodii/exodii_Luliya.json:57]:
```suggestion
      "no": "[You see a tall golden figure with female features.  Strapped to the back of the head with copper wire is a glass cylinder containing long frizzy black hair.  The body stands still and emits an unpleasant beeping sound.  On the back there is a large red button with strange markings.  There is a golden harp of some kind.]"
```
She's not in a closet, so this intro probably shouldn't introduce her as being inside a closet.

**Review by MNG-cataclysm — CHANGES_REQUESTED** [review]

File: data/json/npcs/exodii/exodii_Luliya.json [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+        "topic": "TALK_LULIYA_MUSIC2",
+        "condition": { "u_has_item": "violin" }
+      },
+      {
+        "text": "[HAS HARMONICA] I happen to have a nice harmonica, want to play with me?",
+        "topic": "TALK_LULIYA_MUSIC2",
+        "condition": { "u_has_item": "harmonica" }
+      },
+      {
+        "text": "[HAS COWBELL] I happen to have a nice cowbell, want to play with me?",
```
Comment (Maleclypse) [inline data/json/npcs/exodii/exodii_Luliya.json:213]: `"I have a fever and the only cure is cowbell, want to play with me?"`

**Review by Maleclypse — COMMENTED** [review]
This is really cool, I like it a lot!

**Review by MNG-cataclysm — APPROVED** [review]

**Pull request merged**